README.md

```diff
@@ -249,6 +249,44 @@ server.on('upgrade', function upgrade(request, socket, head) {
 server.listen(8080);
 ```
 
+### Client authentication
+
+```js
+const http = require('http');
+const WebSocket = require('ws');
+const url = require('url');
+
+const server = http.createServer();
+const wss = new WebSocket.Server({ noServer: true });
+
+wss.on('connection', function(ws, request, client) {
+  ws.on('message', function(message) {
+    console.log(`WS message ${message} from user ${client}`);
+  });
+});
+
+server.on('upgrade', function upgrade(request, socket, head) {
+  try {
+    authenticate(request, (client) => {
```

Comment:
This is a bit weird, as I reader I assume this function works asynchronously as it takes a callback. In that case I would expect it to follow the usual err first pattern and to not throw an error synchronously. I know it's just an example but people really like to copy/paste snippets.

Reply:
I was trying to follow your pattern of using callbacks, should I go with something that returns a Promise?

Reply:
Yes I think it is more useful. Thank you.